internal/controllers/nmc_reconciler.go

```diff
@@ -1109,8 +1172,43 @@ func setWorkerSofdepConfig(pod *v1.Pod, modulesLoadingOrder []string) error {
 	return nil
 }
 
+func addFirmwareCopyCommands(pod *v1.Pod, firmwareHostPath *string, moduleConfig *kmmv1beta1.ModuleConfig) error {
+
+	container, _ := podcmd.FindContainerByName(pod, initContainerName)
+	if container == nil {
+		return errors.New("could not find the worker container")
+	}
+
+	if firmwareHostPath == nil {
+		return errors.New("hostFirmwarePath must be set")
+	}
+
+	const argTmplFirmwareAddition = `
```

Comment:
same as for other init container arguments: it is a short string that can be produced by a simple Sprintf, instead of the whole template flow. Maybe we can create a function that defines all the arguments for init container, instead of splitting between this function and baseWorkerPod?

Reply:
At first I was doing it together but I think it is cleaner when the firmware commands are added inside the firmware section rather than having another `if` block in the base pod to handle firmware addition.

Reply:
Changed to fmt.Sprintf.

Reply:
Fixed.